src/ILVerification/src/TypeVerifier.cs

```diff
@@ -98,30 +99,47 @@ public void VerifyInterfaces()
                         MethodDesc resolvedMethod = type.ResolveInterfaceMethodTarget(method);
                         if (resolvedMethod is null)
                         {
-                            VerificationError(VerifierError.InterfaceMethodNotImplemented, Format(type), Format(implementedInterface.DefType), Format(method));
+                            VerificationError(VerifierError.InterfaceMethodNotImplemented, Format(type), Format(implementedInterface.DefType, implementedInterface.InterfaceImplementationHandle), Format(method));
                         }
                     }
                 }
             }
         }
 
-        // Format helpers for future use
-        // The idea is to improve formatting string with tokens value i.e.: 
-        // [Assembly]Class(0x1234) vs current [Assembly]Class
-        // in case somebody will pass '--tokens' switch to ILVerify
         private string Format(EcmaType type)
         {
-            return type.ToString();
+            if (_verifierOptions.IncludeMetadataTokensInErrorMessages)
+            {
+                return $"{type}(0x{_module.MetadataReader.GetToken(type.Handle)})";
```

Comment:
This is to be Def token (TypeDef, MethodDef or FieldDef). The Def token may be for module being verified or for some other module. The module is not displayed, so you have to guess. It makes the tokens to be not very useful.

Here the options to fix I can think of:
- Display the token module next to the token
- Or display the def token only if it is for the module being verified
- Or track the ref or spec token that the type resolved from. This is hard, and it won't be always possible to do accurately for more complex verification rules.

Thoughts?

Reply:
> The Def token may be for module being verified or for some other module

To be clear, as "module" are you meaning "file name" i.e.: `.module CountDown.exe` page 120 ECMA.

Reply:
Yes,that would work. I do not have a strong opinion about which one out of the multiple possible assembly or module names to use here.